app/Http/Controllers/ProductController.php

```diff
@@ -3,89 +3,99 @@
 namespace App\Http\Controllers;
 
 use App\Models\Product;
+use App\Models\User;
 use Illuminate\Http\Request;
 
 class ProductController extends Controller
 {
-    public function store (Request $request)
+
+    public function store(Request $request, Product $product)
     {
+
         $request->validate([
-            'name'=>['required', 'max: 200'],
+             'name'=>['required', 'max: 200'],
              'description'=>['required', 'max: 200'],
-            'price'=>['required', 'max: 200'],
-             'qty'=>['required', 'max: 200'],
+             'price'=>['required', 'max: 200'],
         ]);
 
-        Product::create($request->all());
+
+        $product->user_id=auth()->user()->id;
+        $product->name=$request->input('name');
+        $product->description=$request->input('description');
+        $product->price=$request->input('price');
+        $product->save();
 
         return response()->json(['message'=>'Product Added Successfully!'],200);
 
     }
 
-    public function index ()
-    {
-        $products=Product::all();
-        return response()->json(['products'=> $products]);
-    }
 
-    public function show($id)
+
+ public function index(Product $product)
+ {
+    $user = User::find(auth()->user()->id);
+    $products = $user->products;
+    return response()->json(['products'=> $products]);
+ }
+
+    public function show (Product $id)
     {
-          $product=Product::find($id);
-          if ($product)
-          {
-          return response()->json(['product'=>$product]);
-          }
-          else
-          {
-            return response()->json(['message'=>'Product Not Found'],404);
-          }
+        if ($id->user_id==auth()->user()->id)
+        {
+             return $id;
+        }
+        else
+        {
+            return response()->json(['message'=>'You have no product with this ID.'],404);
+        }
     }
 
 
-    public function update(Request $request, $id)
+    public function update(Request $request, Product $id)
     {
+
+        if ($id->user_id==auth()->user()->id)
```

Comment:
this part is called Authorization. There are 2 ways to handle authorization. via Gates or Policies.
please read about those and implement accordinlgy.